operator/main.go

```diff
@@ -273,15 +307,26 @@ func newExporter() (trace.SpanExporter, error) {
 	)
 }
 
+func newOTelExporter(env envConfig) (trace.SpanExporter, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 3*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, env.OTelCollectorURL, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create gRPC connection to collector at %s: %w", env.OTelCollectorURL, err)
+	}
+	traceExporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithGRPCConn(conn))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create trace exporter: %w", err)
+	}
+	return traceExporter, nil
+}
+
 func newResource() *resource.Resource {
```

Comment:
previously, the resource attributes were all empty and the resource was set to `null` in all spans. This change seems to solve this